examples/shaded-relief.js

```diff
@@ -0,0 +1,158 @@
+// NOCOMPILE
+goog.require('ol.Map');
+goog.require('ol.View');
+goog.require('ol.layer.Image');
+goog.require('ol.layer.Tile');
+goog.require('ol.source.Raster');
+goog.require('ol.source.TileJSON');
+goog.require('ol.source.XYZ');
+
+
+/**
+ * Generates a shaded relief image given elevation data.  Uses a 3x3
+ * neighborhood for determining slope and aspect.
+ * @param {Array.<ImageData>} inputs Array of input images.
+ * @param {Object} data Data added in the "beforeoperations" event.
+ * @return {Array.<ImageData>} Output images (only the first is rendered).
```

Comment:
On thinking some more about this comment, I decided to simplify things in 7b67a724de80e92b0a02fca4d8db6e86c26fa098 so that only a single operation is run, and that operation returns just a single `ImageData` (for image type operations) or a `ol.raster.Pixel`.  The `lib` option allows operations to call other functions, and this is a more flexible way of solving what I was originally after with the operation pipeline.